src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_StringConcat.cs

```diff
@@ -151,24 +159,43 @@ private void FlattenConcatArg(BoundExpression lowered, ArrayBuilder<BoundExpress
         /// its args if so.
         /// </summary>
         /// <returns>True if this is a call to a known string concat operator, false otherwise</returns>
-        private bool TryExtractStringConcatArgs(BoundExpression lowered, out ImmutableArray<BoundExpression> arguments)
+        private bool TryExtractStringConcatArgs(BoundExpression lowered, out ImmutableArray<BoundExpression> arguments, out ImmutableArray<LocalSymbol> previousLocals)
         {
-            switch (lowered.Kind)
+            previousLocals = default;
+
+            switch (lowered)
             {
-                case BoundKind.Call:
-                    var boundCall = (BoundCall)lowered;
+                case BoundCall boundCall:
                     var method = boundCall.Method;
                     if (method.IsStatic && method.ContainingType.SpecialType == SpecialType.System_String)
                     {
-                        if ((object)method == (object)_compilation.GetSpecialTypeMember(SpecialMember.System_String__ConcatStringString) ||
-                            (object)method == (object)_compilation.GetSpecialTypeMember(SpecialMember.System_String__ConcatStringStringString) ||
-                            (object)method == (object)_compilation.GetSpecialTypeMember(SpecialMember.System_String__ConcatStringStringStringString))
+                        if ((object)method == _compilation.GetSpecialTypeMember(SpecialMember.System_String__ConcatStringString) ||
```

Comment:
>(object)method == _compilation.GetSpecialTypeMember(SpecialMember.System_String__ConcatStringString) || [](http://example.com/codeflow?start=28&length=103)

Are changes to this an two following lines style only? I recommend reverting them then.